File: stu_spring/webflux/test/auth/src/main/java/com/webflux/auth/domain/user/facade/UserFacadeImpl.java
package com.webflux.auth.domain.user.facade;

import com.webflux.auth.domain.user.entity.User;
import com.webflux.auth.domain.user.entity.UserRepository;
import com.webflux.auth.domain.user.exception.UserNotFoundException;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;
import reactor.core.publisher.Mono;

@RequiredArgsConstructor
@Component
public class UserFacadeImpl implements UserFacade {

    private final UserRepository userRepository;

    @Override
    public Mono<User> getUser(String email) {
        return userRepository.findByEmail(email)
                .switchIfEmpty(Mono.error(UserNotFoundException::new));
    }
}

File: kotlin/kotlin_basic/src/main/kotlin/com/kotlin/test/kotlintest/domain/user/entity/UserFacadeImpl.kt
package com.kotlin.test.kotlintest.domain.user.entity

import com.kotlin.test.kotlintest.domain.user.exceptions.UserNotFoundException
import org.springframework.data.repository.findByIdOrNull
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.stereotype.Component

@Component
class UserFacadeImpl(
    private val userRepository: UserRepository
) : UserFacade {

    override fun getUserById(id: Long): User {
        return userRepository.findByIdOrNull(id) ?: throw UserNotFoundException.EXCEPTION
    }

    override fun getCurrentUser(): User {
        val id = getUsername()
        return userRepository.findByIdOrNull(id) ?: throw UserNotFoundException.EXCEPTION
    }

    private fun getUsername(): Long {
        val authentication = SecurityContextHolder.getContext()
            .authentication
        return authentication.name.toLong()
    }

}